addons/mail/models/mail_message.py

```diff
@@ -1052,15 +1052,21 @@ def message_format(self, format_reply=True, msg_vals=None):
             notifs = message_sudo.notification_ids.filtered(lambda n: n.res_partner_id)
             vals.update({
                 'needaction_partner_ids': notifs.filtered(lambda n: not n.is_read).res_partner_id.ids,
+                'starredPersonas': [{'id': id, 'type': "partner"} for id in message_sudo.starred_partner_ids.ids],
                 'history_partner_ids': notifs.filtered(lambda n: n.is_read).res_partner_id.ids,
                 'is_note': message_sudo.subtype_id.id == note_id,
                 'is_discussion': message_sudo.subtype_id.id == com_id,
                 'subtype_description': message_sudo.subtype_id.description,
                 'recipients': [{'id': p.id, 'name': p.name, 'type': "partner"} for p in message_sudo.partner_ids],
                 'scheduledDatetime': scheduled_dt_by_msg_id.get(vals['id'], False),
             })
-            if vals['model'] and self.env[vals['model']]._original_module:
-                vals['module_icon'] = modules.module.get_module_icon(self.env[vals['model']]._original_module)
+            if vals['model'] and vals['res_id']:
+                originThread = {'model': vals['model'], 'id': vals['res_id']}
+                if vals['model'] != 'discuss.channel':
+                    originThread.update({'name': vals['record_name']})
```

Comment:
`update` is not often used, especially for single line just `originThread["name"] =` is easier